Language/Functions/Communication/Serial/readBytesUntil.adoc

```diff
@@ -49,7 +49,7 @@ Data type: `size_t`.
 
 [float]
 === Notes and Warnings
-The terminator character is discarded from the serial buffer.
+The terminator character is discarded from the serial buffer, unless the return value equals `length`. If the return value is less than `length`, it cannot be determined whether this is due to a time-out or due to the detection of the supplied terminator. This function relies on millis() which uses timer0's overflow interrupt.
```

Comment:
```
unless the return value equals `length`. 
```
I think this is a reasonable clarification to add to the documentation, but the approach of talking about the return value made it a bit hard for me to understand when I put my "beginner glasses" on. I think it can be reworded to be more clear.

What do you think about this:
>unless the number of characters read equals `length`.

---
```
If the return value is less than `length`, it cannot be determined whether this is due to a time-out or due to the detection of the supplied terminator.
```
I feel that this information is already provided by the main description:
>The function terminates if the terminator character is detected, the determined length has been read, or it times out

---
```
This function relies on millis() which uses timer0's overflow interrupt.
```
Is this true for all architectures officially supported by Arduino?:
- [AVR](https://github.com/arduino/ArduinoCore-avr)
- [megaAVR](https://github.com/arduino/ArduinoCore-megaavr)
- [SAM](https://github.com/arduino/ArduinoCore-sam)
- [SAMD](https://github.com/arduino/ArduinoCore-samd)
- [MBED](https://github.com/arduino/ArduinoCore-nRF528x-mbedos)
- [ARC32](https://github.com/arduino/ArduinoCore-arc32)

We need to be careful to not assume that Arduino == AVR because it's much more than that. However, we don't need to worry about covering 3rd party cores in the Arduino Language Reference. It is the responsibility of their authors to document where there are differences from the official Arduino documentation.